text/0000-stabilize-arbitrary-self-types.md

```diff
@@ -240,56 +216,46 @@ For this reason, implementing `Deref::deref` is problematic for nearly everyone
+Moreover, the blanket implementation constrains this RFC to be only a small delta from the existing unstable "arbitrary self types" mechanism, which is based around `Deref` and has proven to be useful. Separating `Receiver` behavior from `Deref` is a more substantial and riskier evolution to the Rust language.
 
-However, we do not propose to stabilize this. Firstly because of the concerns mentioned in the [method Shadowing section](#method-shadowing). Secondly because we do not want to encourage the use of raw pointers, but rather that raw pointers are wrapped in a custom smart pointer that encodes and documents the invariants.
+(A further suggestion had been to provide separate `Receiver` and `Deref` traits yet have the method resolution logic explore both. This seems to offer no advantages over the blanket implementation, and gives a worst-case O(n*m) number of method candidates).
 
-## Enable for pointers with additional diagnostics
+## Generic parameter
 
-Elsewhere in Rust, there are already diagnostics saying
-> a method with this name may be added to the standard library in the future and warning about the consequences.
+Change the trait definition to have a generic parameter instead of an associated type. There might be permutations here which could allow a single smart pointer type to dispatch method calls to multiple possible receivers - but this would add complexity, no known use case exists, and it might cause worst-case O(n^2) performance on method lookup.
 
-If we chose to stabilize arbitrary self types for raw pointers too, we could simply warn that _any_ use of a raw pointer as a self type could be subject to future shadowing by standard library. But, that would essentially make this feature perpetually warny for raw pointers, so it seems better to just remove it.
+## Do not enable for pointers
 
-## Enable for pointers behind an unstable flag
+It would be possible to respect the `Receiver` trait without allowing dispatch onto raw pointers - they are essentially independent changes to the candidate deduction algorithm.
 
-The current unstable `arbitrary_self_types` feature _does_ allow method dispatch on raw pointers. For anyone relying on that, this is a breakage. We could add that facility behind an alternative unstable flag.
+We don't want to encourage the use of raw pointers, and would prefer rather that raw pointers are wrapped in a custom smart pointer that encodes and documents the invariants. So, there's an argument not to stabilize the raw pointer support.
 
-However, as discussed under [method Shadowing section](#method-shadowing) this would prevent Rust from ever adding more methods to the raw pointer primitive type. That doesn't feel like it will be OK, and so there is no known path to stabilizing raw pointer self types. Therefore, we choose to just remove this facility.
+However, they're tied together in the existing unstable `arbitrary_self_types` feature so on balance it makes sense to stabilize them together.
```

Comment:
I am not sure I get this argument

Reply:
OK. I have replaced it with:

> 
However, the current unstable `arbitrary_self_types` feature provides support for raw pointer receivers, and with years of experience no major concerns have been spotted. We would prefer not to deviate from the existing support more than necessary. Moreover, we are led to believe that raw pointer receivers are quite important for the future of safe Rust, because stacked borrows makes it illegal to materialize references in many positions, and there are a lot of operations (like going from a raw pointer to a raw pointer to a field) where users don't need to or want to do that. We think the utility of including raw pointer receivers outweighs the risks of tempting people to over-use raw pointers.